ПР2

AriaTod opened the pull request.

AriaTod committed:
+ index.html

Каркас

AriaTod committed:
upd index.html + script.js

Был обновлен файл index.html. Там появилась ссылка на файл по js. js-файл был залит тестовый

AriaTod committed:
fix bags

Локально все работает, но script.js не подгружается

AriaTod committed:
upd script.js

Код создаёт страницу: шапку, форму добавления задач, панель фильтров и список.

AriaTod committed:
end-fix bag with display

AriaTod committed:
end-fix bag with file

AriaTod committed:
Delete Index.html

AriaTod committed:
Delete index.html.html

AriaTod committed:
Delete inedx.html

AriaTod committed:
pj rabotai)

AriaTod committed:
upd script.js

Добавил логику добавления, удаления, редактирования и отметки задач.

AriaTod committed:
upd script.js

Добавлена фильтрация задач по статусу, сортировка задач по дате, поиск по названию, универсальная функция "renderTasks(displayTasks)"

AriaTod committed:
upd script.js

+ LocalStorage

AriaTod committed:
upd script.js

+ drag-and-drop

AriaTod committed:
+ css

AriaTod committed:
last version

fix bags and compare with tasks

AriaTod committed:
update for ToDo

+ check out on date

AriaTod committed:
Update README.md

AriaTod committed:
upd file (fix bags)

- innerhtml

AriaTod committed:
Update README.md

Dasha2101 commented on README.md:
```diff
@@ -1,3 +1,49 @@
+- Исправлены ошибки при рендере.
+- (+) валидация даты при редактировавнии.
+- Проведена проверка корректности работы фильтров и сохранения данных.
+- (-) innerHTML
```
Хороший отчет

Dasha2101 commented on index.html:
```diff
@@ -0,0 +1,13 @@
+<html lang="ru">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
```
Не вижу у вас фавикона

Dasha2101 commented on script.js:
```diff
@@ -0,0 +1,350 @@
+// СОЗДАНИЕ ОСНОВНОЙ СТРУКТУРЫ
+// ====================
+
+const app = document.createElement("div");
```
Чтобы избежать дублирования в постоянном создании одних и тех же элементов, можно предусмотреть следующее:
 

export function createButton({ text, className = "", onClick }) { const button = document.createElement("button"); button.textContent = text; button.className = btn ${className}; if (onClick) button.addEventListener("click", onClick); return button; }

USE:
import { createButton } from "./Button.js";
const saveBtn = createButton({ text: "Сохранить", className: "save-btn", onClick: onSave, });

Dasha2101 replied:
Тоже самое касается других элементов

Dasha2101 commented on script.js:
```diff
@@ -0,0 +1,350 @@
+app.appendChild(main);
+
+// --------------------
+// ФОРМА ДОБАВЛЕНИЯ ЗАДАЧ
```
Хороший момент с комментариями, так проверяющему легче понять, где у вас что

Dasha2101 commented on script.js:
```diff
@@ -0,0 +1,350 @@
+  if (newTitle === null) return;
+  const trimmedTitle = newTitle.trim();
+  if (trimmedTitle === "") {
+    alert("Название задачи не может быть пустым!");
```
Уберите пожалуйста alert, promt, confrim из кода, на прошлых парах предупреждала об этом :)

Dasha2101 commented on script.js:
```diff
@@ -0,0 +1,350 @@
+  }
+
+  const currentDate = task.date || "";
+  const newDateInput = prompt("Измените дату (ДД.ММ.ГГГГ):", currentDate);
```
the same

Dasha2101 commented on script.js:
```diff
@@ -0,0 +1,350 @@
+  const trimmedDate = newDateInput.trim();
+
+  if (trimmedDate !== "" && !isValidDisplayDate(trimmedDate)) {
+    alert("Некорректная дата! Используйте формат: ДД.ММ.ГГГГ (например: 05.12.2025)");
```
the same

Dasha2101 commented on script.js:
```diff
@@ -0,0 +1,350 @@
+  const dateInput = inputDate.value.trim();
+
+  if (title === "") {
+    alert("Пожалуйста, введите название задачи!");
```
the same

Dasha2101 commented on script.js:
```diff
@@ -0,0 +1,350 @@
+  }
+
+  if (dateInput !== "" && !isValidDisplayDate(dateInput)) {
+    alert("Некорректная дата! Используйте формат: ДД.ММ.ГГГГ (например: 05.12.2025)");
```
the same

Dasha2101 reviewed (commented):
Добрый день!

Все комментарии указала в пр. По поводу работы, пункты все выполнены, супер. Однако, исправьте, пожалуйста, следующие пункты, чтобы я приняла работу:
1) Фавикон
2) Убрать из кода alert, promt 
Предупреждала на парах об этом, что больше это не используем :) 

Буду ждать ваших доработок, баллы за доработки не снижаются. 

Успехов далее!

AriaTod committed:
Правки по комментариям

- Добавил фавикон.
- Убрал promt и alert (редактирование задачи происходит через contentEditable).
